### ISSUE TITLE
provider/azure: Some changes to azurerm_network_interface are ignored

### ISSUE THREAD

**Issue by Bowbaq** [hashicorp/terraform#6796]
### Terraform Version

Terraform v0.7.0-dev (84a0a32b1947f52deeac344ec5e49393e8a432e1)
### Affected Resource(s)
- azurerm_network_interface
### Terraform Configuration Files

``` hcl
resource "azurerm_network_interface" "public-slave" {
  name                = "${var.name}-public-slave-${count.index}-nic"
  location            = "${var.location}"
  resource_group_name = "${azurerm_resource_group.dcos.name}"

  ip_configuration {
    name      = "public-slave-${count.index}"
    subnet_id = "${azurerm_subnet.public-slave.id}"

    private_ip_address_allocation = "dynamic"
    load_balancer_backend_address_pools_ids = ["${azurerm_simple_lb.public-slave.backend_pool_id}"]
  }

  count = 2
}
```
### Expected Behavior

Adding the `load_balancer_backend_address_pools_ids` key should trigger an update or the network interface
### Actual Behavior

The change of configuration is not detected by Terraform
### Steps to Reproduce

Please list the steps required to reproduce the issue, for example:
1. Create a network interface, without specifying `load_balancer_backend_address_pools_ids`
2. Add a value for `load_balancer_backend_address_pools_ids` and run `terraform plan`
### References

I believe the problem is the hash function for the sub-resource: https://github.com/hashicorp/terraform/blob/156d2916d8f4f8eeeb240e6f565edd40d521ff93/builtin/providers/azurerm/resource_arm_network_interface_card.go#L326-L334

Some of the keys are hashed, but not all of them. I'm not sure why that is. Rewriting it as the following seems to fix the diff problem, but might have undesirable side-effects:

``` go
func resourceArmNetworkInterfaceIpConfigurationHash(v interface{}) int {
    var buf bytes.Buffer
    m := v.(map[string]interface{})
    buf.WriteString(fmt.Sprintf("%s-", m["name"].(string)))
    buf.WriteString(fmt.Sprintf("%s-", m["subnet_id"].(string)))
    buf.WriteString(fmt.Sprintf("%s-", m["private_ip_address"].(string)))
    buf.WriteString(fmt.Sprintf("%s-", m["private_ip_address_allocation"].(string)))
    buf.WriteString(fmt.Sprintf("%s-", m["public_ip_address_id"].(string)))
    buf.WriteString(fmt.Sprintf("%s-", m["load_balancer_backend_address_pools_ids"].(*schema.Set).GoString()))
    buf.WriteString(fmt.Sprintf("%s-", m["load_balancer_inbound_nat_rules_ids"].(*schema.Set).GoString()))

    return hashcode.String(buf.String())
}
```

**Comment by stack72** [comment]
Hi @Bowbaq 

I have already submitted 1 update to this today :) https://github.com/hashicorp/terraform/pull/6790

This will help the majority of TF users right now. I haven't included the load_balancer as TF doesn't manage those yet

P.

**Comment by Bowbaq** [comment]
Even if it's not currently managed by TF, I think it should be included. It could be passed in as a variable, or be provisioned through the CLI with `local-exec`. In my case, I've merged in https://github.com/hashicorp/terraform/pull/6429 (although that doesn't seem to be quite working)

**Comment by stack72** [comment]
Hi @Bowbaq 

I will try and get this updated now to work with the load balancer stuff. I don't have any load balancers to test this with though (thus why i left it out of this update)

P.

**Comment by enieuw** [comment]
I use the following diff in combination with #6429 and #6467 

```diff
diff --git a/builtin/providers/azurerm/resource_arm_network_interface_card.go b/builtin/providers/azurerm/resource_arm_network_interface_card.go
index 174afc5..b055c84 100644
--- a/builtin/providers/azurerm/resource_arm_network_interface_card.go
+++ b/builtin/providers/azurerm/resource_arm_network_interface_card.go
@@ -345,6 +345,11 @@ func resourceArmNetworkInterfaceIpConfigurationHash(v interface{}) int {
                buf.WriteString(fmt.Sprintf("%s-", m["public_ip_address_id"].(string)))
        }

+       pools := m["load_balancer_backend_address_pools_ids"].(*schema.Set).List()
+       for _, p := range pools {
+               buf.WriteString(fmt.Sprintf("%s-", p.(string)))
+       }
+
        return hashcode.String(buf.String())
 }```

Perhaps this could be included in one of the open load balancer pull requests as it's not really needed without the load balancer functionality.

**Comment by pmcatominey** [comment]
@Bowbaq those properties are written to the hash now if present, should resolve your issue :).

**Comment by stack72** [comment]
Closed via #6429
